src/remote.rs

```diff
@@ -50,6 +50,14 @@ pub struct PushOptions<'cb> {
     pb_parallelism: u32,
 }
 
+/// Holds callbacks for a connection to a `Remote`. Disconnects when dropped
+#[allow(dead_code)]
```

Comment:
Oh you should be able to solve that by prefixing the field names with an underscore (e.g. `_proxy` and `_callbacks`)